### PR TITLE
Cube Build fix if no valid data on array 

### LINKED ISSUE #7989: Variable unpacking error when running `CubeBuildStep`
When running the `CubeBuildStep` with the latest version of the pipeline (version `1.12.1`), I sometimes get the following error:

```
...
  File "/lustre/alice3/data/nemesis/jwst/scripts/oliver/pipelines_dev/pipeline.py", line 757, in reduction_spec3_fn
    Spec3Pipeline.call(
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/stpipe/step.py", line 652, in call
    return instance.run(*args)
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/stpipe/step.py", line 478, in run
    step_result = self.process(*args)
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/jwst/pipeline/calwebb_spec3.py", line 240, in process
    result = self.cube_build(result)
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/stpipe/step.py", line 478, in run
    step_result = self.process(*args)
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/jwst/cube_build/cube_build_step.py", line 373, in process
    result, status = thiscube.build_ifucube()
  File "/home/o/ortk2/miniconda3/envs/py310/lib/python3.10/site-packages/jwst/cube_build/ifu_cube.py", line 619, in build_ifucube
    coord1, coord2, corner_coord, wave, dwave, flux, err, slice_no, rois_pixel, \
ValueError: not enough values to unpack (expected 15, got 13)
```

This appears to be caused by a mismatch between the tuple length returned by `map_detector_to_outputframe` when `num_good == 0` and the tuple length expected in `build_ifucube`:

https://github.com/spacetelescope/jwst/blob/214471023d243f13fa8264b65d29c509973b5907/jwst/cube_build/ifu_cube.py#L1575-L1579

https://github.com/spacetelescope/jwst/blob/214471023d243f13fa8264b65d29c509973b5907/jwst/cube_build/ifu_cube.py#L615-L621

I encountered this error when running `Spec3Pipeline` for MIRI MRS data with the arguments `'steps': {'extract_1d': {'skip': True}, 'cube_build': {'output_type': 'band', 'coord_system': 'ifualign'}}`.

### PR DESCRIPTION
**Pull request by jemorrison**
Bug reported in help desk ticket. 

Closes #7989 

This PR returns the correct number of array elements if no valid data is found on the detector. All the array elements are None. 

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/993/
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by jemorrison** [comment]
Going to run regression tests and waiting for more information from help desk user. I would like to test this on the data they encountered the problem.

**Comment by jemorrison** [comment]
All regression tests ran successfully.
This PR is ready to review

File: CHANGES.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ cube_build
 ----------
 
 - Keep data models opened by cube build open until the step completes. [#7998]
+- Add return x_det and y_det (set to None) when no data is found on detector to build cube. [#8001]
```
Comment (hbushouse) [inline CHANGES.rst:13]: Needs to move up to unreleased section, now that 1.12.4 has been released.

Comment (hbushouse) [inline CHANGES.rst:13]:
```suggestion
- Return x_det and y_det set to None when no data is found on detector for building cube. [#8001]
```

**Review by hbushouse — APPROVED** [review]
Code updates looks OK. Starting another regtest run, because it's been a long time since the previous one.

**Comment by hbushouse** [comment]
New regtest run: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1057

**Comment by hbushouse** [comment]
Regtest run looks good.

**Pull request merged**